symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/signal/SignalService.java

```diff
@@ -0,0 +1,163 @@
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public List<Signal> listSignals(Integer skip, Integer limit) {
+    return executeAndRetry("listSignals",
+        () -> signalsApi.v1SignalsListGet(authSession.getSessionToken(), authSession.getKeyManagerToken(), skip, limit));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Stream<Signal> listSignalsStream(Integer chunkSize, Integer totalSize) {
+    PaginatedApi<Signal> api = ((this::listSignals));
+
+    final int actualChunkSize = chunkSize == null ? 50 : chunkSize;
+    final int actualTotalSize = totalSize == null ? 50 : totalSize;
+
+    return new PaginatedService<>(api, actualChunkSize, actualTotalSize).stream();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Signal getSignal(String id) {
+    return executeAndRetry("getSignal",
+        () -> signalsApi.v1SignalsIdGetGet(authSession.getSessionToken(), id, authSession.getKeyManagerToken()));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Signal createSignal(BaseSignal signal) {
+    return executeAndRetry("createSignal",
+        () -> signalsApi.v1SignalsCreatePost(authSession.getSessionToken(), signal, authSession.getKeyManagerToken()));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Signal updateSignal(String id, BaseSignal signal) {
+    return executeAndRetry("updateSignal",
+        () -> signalsApi.v1SignalsIdUpdatePost(authSession.getSessionToken(), id, signal, authSession.getKeyManagerToken()));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void deleteSignal(String id) {
+    executeAndRetry("deleteSignal",
+        () -> signalsApi.v1SignalsIdDeletePost(authSession.getSessionToken(), id, authSession.getKeyManagerToken()));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ChannelSubscriptionResponse subscribeSignal(String id, Boolean pushed, List<Long> userIds) {
+    return executeAndRetry("subscribeSignal",
+        () -> signalsApi.v1SignalsIdSubscribePost(authSession.getSessionToken(), id, authSession.getKeyManagerToken(), pushed, userIds));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ChannelSubscriptionResponse unsubscribeSignal(String id, List<Long> userIds) {
+    return executeAndRetry("unsubscribeSignal",
+        () -> signalsApi.v1SignalsIdUnsubscribePost(authSession.getSessionToken(), id, authSession.getKeyManagerToken(), userIds));
+  }
+
+  /**
+   * {@inheritDoc}
+   * @return
+   */
+  @Override
+  public List<ChannelSubscriber> subscribers(String id, Integer skip, Integer limit) {
+    return executeAndRetry("subscribers",
+        () -> signalsApi.v1SignalsIdSubscribersGet(authSession.getSessionToken(), id, authSession.getKeyManagerToken(), skip, limit)).getData();
+  }
+
+  /**
+   * {@inheritDoc}
+   * @return
+   */
+  @Override
+  public Stream<ChannelSubscriber> subscribersStream(String id, Integer chunkSize, Integer totalSize) {
```

Comment:
passing nulls when you have no ideas of the chunkSize/totalSize to use is not very pretty in term of API
One solution could to have a Pagination object (that comes with default values)
Or to have different methods with/without pagination

Reply:
I'm following the paginating strategy which was implemented in [MessageService](https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/blob/master/symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java#L369)
But I think having different methods with/without pagination should be consider.